### PR TITLE
Type rename provider [Uses VS Code proposed API]

### PR DESCRIPTION
**Pull request by xorye**
Follows HTML's implementation: https://github.com/microsoft/vscode/commit/01e01b13f88064374ebc5aa789cb0ee76d03a17c

To test this PR, you need the server side PR  as well https://github.com/eclipse/lemminx/pull/836

Signed-off-by: David Kwon <dakwon@redhat.com>

### CODE REVIEW

**Comment by xorye** [comment]
Currently not working, but you need to use VS Code insiders https://code.visualstudio.com/insiders/ and set `editor.renameOnType` to true.

Demo for HTML: https://github.com/microsoft/vscode/issues/88424#issuecomment-600840619

**Comment by xorye** [comment]
It works now:
onTypeRename

But we still suffer from this issue: https://github.com/redhat-developer/vscode-xml/pull/213#issuecomment-579328288
onTypeRenameCompletion

File: package.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
   "license": "EPL-1.0",
   "bugs": "https://github.com/redhat-developer/vscode-xml/issues",
   "preview": true,
+  "enableProposedApi": true,
```
Comment (fbricon) [inline package.json:12]: we can only use it in insiders. The marketplace rejects publishing when enableProposedApi:true

**Comment by angelozerr** [comment]
> But we still suffer from this issue: #213 (comment)

It's very strange because this problem should be fixed. The only problem that I have seen is when you do this usecase with a XML element which doesnt comes from a XSD Schema (but it seems that it's not our usecase).

In the demo, I noticed that when you remove the d character completion opens, inmy case the completion is not opened (I must do a Ctrl+Space to open completion). Is there a settings for that?

**Comment by angelozerr** [comment]
We need to wait for https://github.com/microsoft/vscode/issues/105394

File: package.json (hunk 1 above)
Comment (angelozerr) [reply package.json:12]: We need to wait for microsoft/vscode#105394

**Comment by angelozerr** [comment]
Please note the onTypeRanme is now in the LSP 3.16 specification. I have created an issue in LSP4J https://github.com/eclipse/lsp4j/issues/469 If I understand correctly, we will  avoid developping TypeScript code for supporting mirror cursor.

**Comment by angelozerr** [comment]
Today this type rename provider became linked editing range with a LSP specification. This PR is replaced with https://github.com/redhat-developer/vscode-xml/pull/432

**Pull request closed without merging**